docs/proposals/policy.md

```diff
@@ -0,0 +1,389 @@
+		"name": "admin"
+	}
+	"userNames": ["Hubert"]
+}
+```
+
+`Hubert` now has the power to manage policy on the `chisel` project.  He can grant permission to users who should have access.  In this case, he grants edit permission to `Curt` using the predefined role.
+```
+{
+	"kind": "roleBinding",
+	"name": "Editors",
+	"namespace": "chisel",
+	"referencesGlobalRole": true,
+	"roleName": "editor",
+	"userNames": ["Curt"]
+}
+```
+
+That covers most cases where global admins want to create projects and delegate the administration of a project to a project admin.  Then the project admin wants to delegate project level permissions to other users.  We can continue the example with `Hubert` (our project admin) wanting to create custom roles and restrictions.
+
+`Hubert` wants to create a bot that only has permission to touch labels on DeploymentConfigs.  He can create a role like this (note that a role is required because we want to have a set of two rules):
+```
+{
+	"kind": "role",
+	"name": "deploymentConfigLabelers",
+	"namespace": "chisel",
+	"rules": [
+		{ "verbs": ["watch", "list", "get"], "resourceKinds": ["DeploymentConfig"] },
+		{ "verbs": ["update"], "resourceKinds": ["DeploymentConfig"], "attributeRestrictions" : {"fieldsMutatable": ["labels"]} }
+	]
+}
+{
+	"kind": "roleBinding",
+	"name": "DeploymentConfigLabelerBots",
+	"namespace": "chisel",
+	"roleRef": {
+		"namespace": "chisel",
+		"name": "deploymentConfigLabelers"
+	}
+	"userNames": ["ProtectorBot", "DeprotectorBot"]
+}
+```
+
+Finally, `Hubert` wants to prevent `Curt` from ever deleting a protected deploymentConfig again, so he creates this role and roleBinding:
+```
+{
+	"kind": "role",
+	"name": "fatFingeredEditor",
+	"namespace": "chisel",
+	"rule": [
+		{ "Deny": "true", "Verbs": ["delete"], "ResourceKinds": ["DeploymentConfig"], "AttributeRestrictions" : {"labelsContain": ["protected"]} }
+	]
+}
+{
+	"kind": "roleBinding",
+	"name": "FatFingeredEditors",
+	"namespace": "chisel",
+	"roleRef": {
+		"namespace": "chisel",
+		"name": "fatFingeredEditor"
+	}
+	"userNames": ["Curt"]
+}
+```
+
+## Basic Concepts
+ 1.  A PolicyRule expresses a permission containing: Deny, []Verb, []ResourceKinds (pod, deploymentConfig, *-(all kinds), etc)
+ 1.  A Role is a way to name a set of PolicyRules
+ 1.  A RoleBinding is a way to associate a single PolicyRule with a given user or group.  A RoleBinding references (but does not include) a Role in either the local namespace or the global namespace.
+ 1.  A global namespace exists.  This namespace is special.  Roles defined in the global namespace can be referenced from any other namespace.  Bindings defined in the global namespace apply to **all** namespaces (an admin in the global namespace can edit any resource in any namespace).  RoleBinding policy rules cannot be overridden by any local binding (a local PolicyBindingRule that denies actions to the global admin will be overriden by the binding defined in the global namespace).
```

Comment:
If you do this, then policies aren't really namespaced.  Instead, you're trying to have policies in a namespace and outside a namespace.  I would strongly recommend simply having policies not be namespaced (or the policy engine uses only the policies in a namespace) and allow role bindings to exist, one per namespace (or have a convention they are named per namespace).  I don't think you want the fundamental policy stored in etcd as well - namespaces aren't really hierarchal, but what you're trying to create is a hierarchy.